Cargo.toml

```diff
@@ -46,10 +46,23 @@ thiserror = "1.0.29"
 url = "2.2.2"
 webbrowser = { version = "0.8.0", optional = true }
 
+[target.'cfg(target_arch = "wasm32")'.dependencies]
+getrandom = { version = "0.2.3", features = ["js"] }
+
+[target.'cfg(not(target_arch = "wasm32"))'.dependencies]
+getrandom = "0.2.3"
+
 [dev-dependencies]
 env_logger = { version = "0.11.0", default-features = false }
-tokio = { version = "1.11.0", features = ["rt-multi-thread", "macros"] }
 futures-util = "0.3.17"
+wasm-bindgen-test = "0.3.34"
+
+[target.'cfg(not(target_arch = "wasm32"))'.dev-dependencies]
+tokio = { version = "1.11.0", features = ["rt-multi-thread", "macros"] }
+
+[target.'cfg(target_arch = "wasm32")'.dev-dependencies]
+tokio = { version = "1.11.0", features = ["rt", "macros"] }
+dotenvy_macro = { version = "0.15.7" }
```

Comment:
Used to read `.env` when running wasm tests

Reply:
Correct, `std::env::var` won't work on `wasm32-unknown-unknown`.Wasm is designed to run in a web browser which is isolated from the host OS. 

From my understanding of cargo the `#[cfg...]` on line 63 creates a block section, so it should already be imported only for wasm. It doesn't seem to get built when running `cargo build --tests`:
```
% cargo build --tests
   Compiling wasm-bindgen-backend v0.2.91
   Compiling tokio v1.35.1
   Compiling futures-util v0.3.30
   Compiling serde v1.0.195
   Compiling thiserror v1.0.56
   Compiling strum v0.26.1
   Compiling enum_dispatch v0.3.12
   Compiling async-stream v0.3.5
   Compiling wasm-bindgen-macro-support v0.2.91
   Compiling wasm-bindgen-macro v0.2.91
   Compiling wasm-bindgen v0.2.91
   Compiling futures-executor v0.3.30
   Compiling serde_json v1.0.111
   Compiling serde_urlencoded v0.7.1
   Compiling chrono v0.4.31
   Compiling futures v0.3.30
   Compiling js-sys v0.3.68
   Compiling console_error_panic_hook v0.1.7
   Compiling tokio-util v0.7.10
   Compiling tokio-native-tls v0.3.1
   Compiling tokio-socks v0.5.1
   Compiling h2 v0.3.23
   Compiling rspotify-model v0.12.0 (/Users/gregory/rust/rspotify/rspotify-model)
   Compiling wasm-bindgen-futures v0.4.41
   Compiling wasm-bindgen-test v0.3.41
   Compiling hyper v0.14.28
   Compiling hyper-tls v0.5.0
   Compiling reqwest v0.11.23
   Compiling rspotify-http v0.12.0 (/Users/gregory/rust/rspotify/rspotify-http)
   Compiling rspotify v0.12.0 (/Users/gregory/rust/rspotify)
    Finished dev [unoptimized + debuginfo] target(s) in 6.82s
    ```

Reply:
For comparison, here's the output with wasm:
```
% cargo build --tests --target wasm32-unknown-unknown
   Compiling proc-macro2 v1.0.76
   Compiling unicode-ident v1.0.12
   Compiling wasm-bindgen-shared v0.2.91
   Compiling bumpalo v3.14.0
   Compiling log v0.4.20
   Compiling autocfg v1.1.0
   Compiling wasm-bindgen v0.2.91
   Compiling cfg-if v1.0.0
   Compiling serde v1.0.195
   Compiling version_check v0.9.4
   Compiling pin-project-lite v0.2.13
   Compiling typenum v1.17.0
   Compiling futures-core v0.3.30
   Compiling itoa v1.0.10
   Compiling tinyvec_macros v0.1.1
   Compiling futures-sink v0.3.30
   Compiling tinyvec v1.6.0
   Compiling syn v1.0.109
   Compiling futures-channel v0.3.30
   Compiling percent-encoding v2.3.1
   Compiling futures-task v0.3.30
   Compiling rustversion v1.0.14
   Compiling pin-utils v0.1.0
   Compiling memchr v2.7.1
   Compiling serde_json v1.0.111
   Compiling generic-array v0.14.7
   Compiling ryu v1.0.16
   Compiling slab v0.4.9
   Compiling futures-io v0.3.30
   Compiling num-traits v0.2.17
   Compiling form_urlencoded v1.2.1
   Compiling unicode-bidi v0.3.14
   Compiling thiserror v1.0.56
   Compiling fnv v1.0.7
   Compiling bytes v1.5.0
   Compiling quote v1.0.35
   Compiling unicode-normalization v0.1.22
   Compiling async-trait v0.1.77
   Compiling heck v0.4.1
   Compiling syn v2.0.48
   Compiling http v0.2.11
   Compiling tower-service v0.3.2
   Compiling idna v0.5.0
   Compiling base64 v0.21.7
   Compiling env_filter v0.1.0
   Compiling dotenvy v0.15.7
   Compiling rspotify-macros v0.12.0 (/Users/gregory/rust/rspotify/rspotify-macros)
   Compiling scoped-tls v1.0.1
   Compiling env_logger v0.11.1
   Compiling url v2.5.0
   Compiling block-buffer v0.10.4
   Compiling crypto-common v0.1.6
   Compiling digest v0.10.7
   Compiling sha2 v0.10.8
   Compiling wasm-bindgen-backend v0.2.91
   Compiling maybe-async v0.2.7
   Compiling dotenvy_macro v0.15.7
   Compiling wasm-bindgen-macro-support v0.2.91
   Compiling serde_derive v1.0.195
   Compiling futures-macro v0.3.30
   Compiling thiserror-impl v1.0.56
   Compiling strum_macros v0.26.1
   Compiling async-stream-impl v0.3.5
   Compiling enum_dispatch v0.3.12
   Compiling wasm-bindgen-test-macro v0.3.41
   Compiling tokio-macros v2.2.0
   Compiling async-stream v0.3.5
   Compiling wasm-bindgen-macro v0.2.91
   Compiling futures-util v0.3.30
   Compiling tokio v1.35.1
   Compiling strum v0.26.1
   Compiling js-sys v0.3.68
   Compiling console_error_panic_hook v0.1.7
   Compiling futures-executor v0.3.30
   Compiling futures v0.3.30
   Compiling wasm-bindgen-futures v0.4.41
   Compiling web-sys v0.3.67
   Compiling getrandom v0.2.12
   Compiling serde_urlencoded v0.7.1
   Compiling chrono v0.4.31
   Compiling wasm-bindgen-test v0.3.41
   Compiling rspotify-model v0.12.0 (/Users/gregory/rust/rspotify/rspotify-model)
   Compiling reqwest v0.11.23
   Compiling rspotify-http v0.12.0 (/Users/gregory/rust/rspotify/rspotify-http)
   Compiling rspotify v0.12.0 (/Users/gregory/rust/rspotify)
    Finished dev [unoptimized + debuginfo] target(s) in 9.77s
 ```